node/node.go

```diff
@@ -124,16 +124,16 @@ func (node *Node) Run() {
 	node.Logger.Info("Shutdown complete!")
 }
 
-func AddPlugin(name string, status int) {
+func AddPlugin(name string, status int, plugin *Plugin) {
```

Comment:
I think it's sufficient and cleaner to only pass `plugin *Plugin` into the function and read the properties `name` and `status` from it